src/main/kotlin/fi/aalto/cs/apluscourses/model/component/ScalaSdk.kt

```diff
@@ -1,54 +1,69 @@
+import org.jetbrains.plugins.scala.project.ScalaLibraryProperties
+import org.jetbrains.plugins.scala.project.ScalaLibraryPropertiesState
+import org.jetbrains.plugins.scala.project.ScalaLibraryType
 import java.nio.file.Path
 
 class ScalaSdk(private val scalaVersion: String, project: Project) : Library(scalaVersion, project) {
+    @NonNls
+    private val versionNumber = scalaVersion.substringAfter("scala-sdk-")
+    private val libPath = Path.of(project.basePath!!, ".libs")
+    private val sdkPath = libPath.resolve("scala3-$versionNumber")
+
     override suspend fun downloadAndInstall(updating: Boolean) {
         status = Status.LOADING
         CoursesLogger.info("Downloading Scala SDK $scalaVersion")
-        val strippedScalaVersion = this.scalaVersion.substringAfter("scala-sdk-")
         val zipUrl =
-            "https://github.com/lampepfl/dotty/releases/download/$strippedScalaVersion/scala3-$strippedScalaVersion.zip"
-        val sourcesUrl = "https://github.com/scala/scala3/archive/refs/tags/$strippedScalaVersion.zip"
-        val path = Path.of(project.basePath!!, ".libs")
+            "https://github.com/lampepfl/dotty/releases/download/$versionNumber/scala3-$versionNumber.zip"
+        val sourcesUrl = "https://github.com/scala/scala3/archive/refs/tags/$versionNumber.zip"
+        val path = libPath
         downloadAndUnzipZip(zipUrl, path)
-        val libraryTable = LibraryTablesRegistrar.getInstance().getLibraryTable(project).modifiableModel
+
+        val libraryTable = libraryTable(project).modifiableModel
         libraryTable.getLibraryByName(scalaVersion)?.let {
             writeAction {
                 libraryTable.removeLibrary(it)
             }
         }
-        val provider = IdeModifiableModelsProviderImpl(project)
-        val library = provider.createLibrary(scalaVersion)
-        val compilerClasspath =
-            path.resolve("scala3-$strippedScalaVersion").resolve("lib").toFile().listFiles().toList()
+        val kind: PersistentLibraryKind<ScalaLibraryProperties> = ScalaLibraryType.`Kind$`.`MODULE$`
+        val library = libraryTable.createLibrary(name, kind)
+        val compilerClasspath = sdkPath.resolve("lib").toFile().listFiles().toList()
         val scala2Version =
             compilerClasspath.find { it.name.startsWith("scala-library") }?.nameWithoutExtension?.substringAfter("scala-library-")
 
         writeAction {
             val libraryModel = library.modifiableModel
-            ScalaSdkUtils.ensureScalaLibraryIsConvertedToScalaSdk(
-                provider,
-                library,
-                Option.apply(strippedScalaVersion.substringBeforeLast(".")),
-                CollectionConverters.asScala(compilerClasspath).toSeq(),
-                CollectionConverters.asScala(listOf<File>()).toSeq(),
-                ScalaSdkUtils.resolveCompilerBridgeJar(strippedScalaVersion)
+
+            // HACK: this is the only way to access properties that I am aware of
```

Comment:
This is from our old code, right? Back to the way we handled it previously?

Reply:
I decided to use the old way for now because the new one only works with either 2024 or 2025, I will ask the IntelliJ Scala team for a better solution